### DISCUSSION TITLE
Queue binding failed，but no exception or error logs

### DISCUSSION THREAD

**Discussion by EricLye** [rabbitmq/rabbitmq-server#9251] [Questions] [answered]
Below is the code.
`
Queue queue = new Queue(queueName, true, false, false, null);
DirectExchange directExchange = new DirectExchange(ACTANKER_DIRECT_EXCHANGE_NAME);
amqpAdmin.declareQueue(queue);
amqpAdmin.declareExchange(directExchange);
amqpAdmin.declareBinding(BindingBuilder.bind(queue).to(directExchange).withQueueName());
`

**Comment by michaelklishin** [comment, marked as answer]
@EricLye there isn't much to work with I'm afraid. We do not guess in this community. 

See [internal events](https://rabbitmq.com/logging.html#internal-events). All new bindings (after you attach an event consumer) should result in events logged there. A [traffic capture](https://rabbitmq.com/amqp-wireshark.html) will show what is actually sent by both the client and the API endpoint it uses.

The client you use may be swallowing exceptions or delaying operations. We don't have any details on what the client is. Putting together an executable way to reproduce would answer this and other questions.

**Reply by EricLye** [reply]
I found reason eventually, it is the code fault. Thanks a lot for your suggestion.
